combat-trainer.lic

```diff
@@ -355,11 +355,25 @@ class LootProcess
   end
 
   def should_perform_ritual(prepare_consume)
-    !prepare_consume &&
-      ((@necro_train_first_aid && DRSkill.getxp('First Aid') < 32) || DRSkill.getxp('Thanatology') < 32)
+   !prepare_consume &&
+    ((@ritual_type == "dissect" && DRSkill.getxp('First Aid') < 32) ||
+    (@ritual_type == "harvest" && DRSkill.getxp('Skinning') < 32) ||
+    DRSkill.getxp('Thanatology') < 32)
+  end
+
+  def check_ritual_xp
+    DRSkill.getxp('First Aid') > 31 && DRSkill.getxp('Skinning') > 31
+  end
+
+  def cycle_rituals
+    @ritual_type = "preserve"
+    @ritual_type = "dissect" if DRSkill.getxp('First Aid') < 32
+    @ritual_type = "harvest" if DRSkill.getxp('Skinning') < DRSkill.getxp('First Aid')
+    @ritual_type = "preserve" if check_ritual_xp
   end
 
   def check_rituals?(game_state)
+    cycle_rituals if @cycle_rituals && !game_state.prepare_consume
```

Comment:
I recommend putting the conditionals inside the `cycle_rituals` function. This would reduce the code in `check_rituals?` and make the flow easier to understand. What do you think?